annif/backend/vw_multi.py

```diff
@@ -91,16 +102,39 @@ def _uris_to_subject_ids(project, uris):
     def _format_examples(self, project, text, uris):
         subject_ids = self._uris_to_subject_ids(project, uris)
         if self.algorithm == 'multilabel_oaa':
-            yield '{} | {}'.format(','.join(map(str, subject_ids)), text)
+            yield '{} {}'.format(','.join(map(str, subject_ids)), text)
         else:
             for subject_id in subject_ids:
-                yield '{} | {}'.format(subject_id + 1, text)
+                yield '{} {}'.format(subject_id + 1, text)
+
+    def _inputs_to_exampletext(self, project, text):
+        namespaces = {}
+        for input in self.inputs:
+            if input == '_text_':
+                normalized = self._normalize_text(project, text)
+                if normalized != '':
+                    namespaces['_text_'] = normalized
+            else:
+                proj = annif.project.get_project(input)
+                result = proj.analyze(text)
+                features = [
+                    '{}:{}'.format(
+                        self._cleanup_text(
+                            hit.uri),
+                        hit.score) for hit in result.hits]
+                namespaces[input] = ' '.join(features)
+        if not namespaces:
+            return None
+        return ' '.join(['|{} {}'.format(namespace, featurestr)
+                         for namespace, featurestr in namespaces.items()])
```

Comment:
The `if not namespaces` check is there not because `self.inputs` could be empty (that would be a stupid configuration although it's not currently an error), but because sometimes none of the inputs would return anything so `namespaces` could remain empty. But I'll see if this can be made more elegant.

Reply:
Ohh, fair point! Hadn't seen that part :+1: